scripts/irods/test/resource_suite.py

```diff
@@ -1167,3 +1167,181 @@ def test_itrim_no_op(self):
+                                   (resource_name, resource_context))
+
+            # create file to put
+            lib.make_file(file1, 100)
+
+            # put small file
+            self.admin.assert_icommand("iput -R %s %s" % (resource_name, file1))  # iput
+
+            # get file
+            self.admin.assert_icommand("iget %s %s" % (file1, file2))  # iget
+
+            # make sure the file that was put and got are the same
+            self.admin.assert_icommand("diff %s %s " % (file1, file2))
+        finally:
+            # local cleanup
+            self.admin.assert_icommand("irm -f " + file1)
+
+            if os.path.exists(file1):
+                os.unlink(file1)
+            if os.path.exists(file2):
+                os.unlink(file2)
+
+            # cleanup
+            lib.remove_resource(resource_name, self.admin)
+
+    def test_detached_mode_use_same_vault_path__issue_4421(self):
+        try:
+            file1 = "f1"
+            file2 = "f2"
+            resource_host = "irods.org"
+            resource_name = 'detached_resource'
+            hostname = lib.get_hostname()
+            resource_context = 'host_mode=detached;host_list=irods.org:/some/path,%s' % hostname
+
+            lib.create_ufs_resource(resource_name, self.admin, resource_host)
+            self.admin.assert_icommand("iadmin modresc %s context %s" %
+                                   (resource_name, resource_context))
+
+            # create file to put
+            lib.make_file(file1, 100)
+
+            # put small file
+            self.admin.assert_icommand("iput -R %s %s" % (resource_name, file1))  # iput
+
+            # get file
+            self.admin.assert_icommand("iget %s %s" % (file1, file2))  # iget
+
+            # make sure the file that was put and got are the same
+            self.admin.assert_icommand("diff %s %s " % (file1, file2))
+        finally:
+            # local cleanup
+            self.admin.assert_icommand("irm -f " + file1)
+
+            if os.path.exists(file1):
+                os.unlink(file1)
+            if os.path.exists(file2):
+                os.unlink(file2)
+
+            # cleanup
+            lib.remove_resource(resource_name, self.admin)
+
+    def test_detached_mode_host_not_in_host_list__issue_4421(self):
+        try:
+            file1 = "f1"
+            resource_host = "irods.org"
+            resource_name = 'detached_resource'
+            resource_context = 'host_mode=detached;host_list=irods.org:/some/path,some_other_host'
+
+            lib.create_ufs_resource(resource_name, self.admin, resource_host)
+            self.admin.assert_icommand("iadmin modresc %s context %s" %
+                                   (resource_name, resource_context))
+
+            # create file to put
+            lib.make_file(file1, 100)
+
+            # put small file
+            self.admin.assert_icommand("iput -R %s %s" % (resource_name, file1), 'STDERR', 'USER_SOCK_CONNECT_ERR')  # iput
```

Comment:
Should we make sure that the data object doesn't exist here as well?

Reply:
done